### PR TITLE
* add support for file

### PR DESCRIPTION
**Pull request by mayitbeegh**

### CODE REVIEW

**Comment by tyrsius** [comment]
I'm a little unclear on this functionality. Are files distinct from values in the cerberus API? They have their own methods in this PR. @fieldju how has this been done in other clients?

**Comment by sdford** [comment]
Hi @tyrsius, if I understand your question correctly..yes, the file API (`v1/secure-file`) is separate from the regular secrets API (`v1/secret`).

**Comment by tyrsius** [comment]
Thanks, thats what I was after.

File: index.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const linereader = require('./lib/linereader')
 const kms = require('./lib/kms')
 const lambda = require('./lib/lambda')
 const sts = require('./lib/sts')
+var FormData = require('form-data')
```
[the remark this answers is not in the document]
Comment (fieldju) [reply index.js]: - [ ] change for consistency

File: index.js [hunk 2]
```diff
@@ -103,6 +112,45 @@ function callCerberus (context, type, keyPath, data, cb) {
   }
 }
 
+function fileCerberus (context, type, filePath, data, cb) {
```
Comment (fieldju) [inline index.js]: - [ ] Add js docs

File: index.js [hunk 3]
```diff
@@ -103,6 +112,45 @@ function callCerberus (context, type, keyPath, data, cb) {
...
+      headers: Object.assign({}, globalHeaders, {'X-Vault-Token': token}, type === 'POST' ? form.getHeaders() : undefined),
+      body: form,
+      json: type === 'LIST' || type === 'DELETE'
+    })
+
+    let fileResult = fileResponse.data
+    if (fileResult && fileResult.errors && fileResult.errors.length > 0) throw new Error(formatCerberusError(fileResult.errors))
+
+    context.log('file retrieved', fileResponse.statusCode.toString(), fileResult)
+    if (fileResponse.statusCode && fileResponse.statusCode.toString()[0] !== '2') throw new Error('File Request error, Status: ' + fileResponse.statusCode)
```
Comment (fieldju) [inline index.js]: How does this not fail a linter?

Please use curly braces for `if statements` and google all the very famous bugs that happen from people not using curly braces on if statements.

https://blog.codecentric.de/en/2014/02/curly-braces/

- [ ] If statements should have curly braces.

Comment (tyrsius) [reply index.js]: Standard allows for un-braced `if` statements if the block is on the same line as the `if` statement. See [the multi option on the curly rule](https://eslint.org/docs/rules/curly)

**Pull request merged**